### ISSUE TITLE
Why textures and geometries need to be created separately?

### ISSUE THREAD

**Issue by shi-yan** [CindyJS/CindyJS#812]
I'm trying to implement an online playground for CindyJS. 

Users can type CindyScript in a text box, and I will run the code and show the result.

But the fact that some resources are created outside the code makes this a bit difficult, for example the textures and geometries. I don't quite understand the reason for geometries being created this way and the document for that is missing. Why can't a point be created with code? Why some geometries are not created this way? What should be created separately? Are those the interactive geometries?

Thanks.

**Comment by kortenkamp** [comment]
There are CindyScript commands that can create points or other elements.  In CindyJS this is not fully implemented, AFAIK, but your request gives rise to check that. 

The reason for this separation is that historically the interactive part of Cinderella has been available since 1997, but CindyScript was introduced in 2003/2004 IIRC. So, we also had interactive geometry objects that were created by user interaction, and there was no need to create them in CindyScript.

**Comment by kortenkamp** [comment]
As an add-on – you know https://cindyjs.org/docs/cindygltutorial/livecoding.html ? This is similar to what you want to do, isn't it?

**Comment by kortenkamp** [comment]
And let me add… you *can* create points there, for example.
<img width="515" alt="image" src="https://user-images.githubusercontent.com/1773555/67304939-e1436a00-f4f4-11e9-92d3-e292075e68e1.png">

**Comment by shi-yan** [comment]
Thank you very much!

One more question, is it possible for javascript to interact with CindyJs? For example, creating a button with html and js, and this button can control a program written in CindyScript?

I'm making a programmable blog (similar to jupyter notebook), https://epiphany.pub/post?refId=2684bc94f9fcb9ffe637ebfbeba2af8c797c6ad9a66181026ee4bd3806b6f211

I hope to adopt CindyJS.

**Comment by kortenkamp** [comment]
yes, it is both possible to execute JavaScript in CindyScript using the `javascript(…)` function, and to execute CindyScript from Javascript using the `evokeCS(…)` function on the Cindy object. There are several examples doing this in the gallery and the examples folder.

**Comment by montaga** [comment]
You can create any geometric element from CindyScript using [create](https://cindyjs.org/ref/Interaction_with_Geometry.html#create$3). There is no CindyScript-method for loading textures. It should be possible to write a 20-line plugin that just does this.

An example where the create command is extensively used is our [prototypical editor](https://cindyjs.org/editor/), which is programmed in [CindyScript](https://cindyjs.org/editor/geometryeditor-plugin.js).
